python/tank/platform/interactive_authentication.py

```diff
@@ -14,172 +14,272 @@
-        login_info.get("login", get_login_name()),
-        config_data.get("http_proxy")
-    )
-
-    logger.debug("Login successful!")
-
-    # Cache the credentials so subsequent session based logins can reuse the session id.
-    authentication.cache_session_data(config_data["host"], login, session_token)
-
-
-def console_renew_session():
-    """
-    Prompts the user to enter his password on the command line to retrieve a new session token.
-    """
-    _do_authentication(_do_console_based_session_renewal)
+    def __init__(self, is_session_renewal):
+        """
+        Creates the UiAuthenticationHandler object.
+        :param is_session_renewal: Indicate if we are renewing a session.
+        """
+        self._is_session_renewal = is_session_renewal
+
+    def _do_authentication(self, hostname, login, http_proxy):
+        """
+        Pops a dialog that asks for the hostname, login and password of the user. If there is a current
+        engine, it will run the code in the main thread.
+        :param hostname: Host to display in the dialog.
+        :param login: login to display in the dialog.
+        :param http_proxy: Proxy server to use when validating credentials. Can be None.
+        :returns: A tuple of (hostname, login, session_token)
+        """
+        from .qt import login_dialog
+
+        from tank.platform import engine
+
+        if self._is_session_renewal:
+            logger.debug("Requesting password in a dialog.")
+        else:
+            logger.debug("Requesting username and password in a dialog.")
+
+        def _process_ui():
+            dlg = login_dialog.LoginDialog(
+                "Shotgun Login",
+                hostname=hostname,
+                login=login,
+                session_renewal=self._is_session_renewal
+            )
+            return dlg.result()
+
+        # If there is a current engine, execute from the main thread.
+        if engine.current_engine():
+            result = engine.current_engine().execute_in_main_thread(_process_ui)
+        else:
+            # Otherwise just run in the current one.
+            result = _process_ui()
+        if not result:
+            raise TankAuthenticationError("No credentials provided.")
+        return result
 
 
 def ui_renew_session():
     """
     Prompts the user to enter his password in a dialog to retrieve a new session token.
     """
-    _do_authentication(_do_console_based_session_renewal)
+    UiAuthenticationHandler(is_session_renewal=True).authenticate()
 
 
 def ui_authenticate():
     """
     Prompts the user to login via a dialog and caches the session token for future reuse.
     """
-    _do_authentication(_do_ui_based_login)
+    UiAuthenticationHandler(is_session_renewal=False).authenticate()
+
+
+def console_renew_session():
+    """
+    Prompts the user to enter his password on the command line to retrieve a new session token.
+    """
+    ConsoleRenewSessionHandler().authenticate()
```

Comment:
Maybe it would make sense to merge `ConsoleLoginHandler` and `ConsoleRenewSessionHandler` ? That would remove one level of inheritance and make it more like the ui based setup!